src/inmanta/server/services/orchestrationservice.py

```diff
@@ -814,9 +820,163 @@ async def _put_version(
+            ) -> None:
+                query = """
+                    SELECT
+                        agent_name,
+                        inmanta_module_name,
+                        inmanta_module_version
+                    FROM
+                        modules_for_agent
+                    WHERE
+                        cm_version=$1
+                    AND
+                        environment=$2
+                 """
+                async with connection.transaction():
+                    values = [partial_base_version, environment]
+                    in_db_module_data: dict[str, dict[str, str]] = defaultdict(dict)
+                    async for record in connection.cursor(query, *values):
+                        in_db_module_data[str(record["inmanta_module_name"])][str(record["agent_name"])] = str(
+                            record["inmanta_module_version"]
+                        )
+
+                for resource_type in type_to_agent.keys():
+                    for agent_name in type_to_agent[resource_type]:
+                        if resource_type not in type_to_module_data:
+                            LOGGER.debug("No inmanta module information was provided for resource type %s." % resource_type)
+                            continue
+                        for module_name in type_to_module_data[resource_type]:
+                            expected_module_version = module_version_info[module_name]["version"]
+
+                            if in_db_module_data[module_name][agent_name] != expected_module_version:
+                                raise BadRequest(
+                                    "Cannot perform partial export because of version mismatch for module %s." % module_name
+                                )
+
+            async def upload_modules(env: uuid.UUID, module_version_info: dict[str, dict[str, Any]]) -> None:
+                if not module_version_info:
+                    return
+                module_stmt = insert(InmantaModule).on_conflict_do_nothing()
+
+                files_in_module_stmt = insert(FilesInModule).on_conflict_do_nothing()
+
+                module_data = []
+                files_in_module_data = []
+                for inmanta_module_name, python_module in module_version_info.items():
+
+                    requirements: set[str] = set()
+
+                    for file in python_module["files_in_module"]:
+                        file_path = file["path"]
+                        match file_path:
+                            case _ if file_path.endswith(".py"):
+                                is_byte_code = False
+                            case _ if file_path.endswith(".pyc"):
+                                is_byte_code = True
+                            case _:
+                                raise ServerError(
+                                    "Invalid file extension for plugin file %s. Expecting `.py` or `.pyc`." % file_path
+                                )
```

Comment:
Why isn't this conversion in the exporter?

And why doesn't it use the existing method  to determine what it is?